### DISCUSSION TITLE
Workflows for re-processing e-book audio generation

### DISCUSSION THREAD

**Discussion by LexiconCode** [DrewThomasson/ebook2audiobook#347] [General]
The end user needs a relationship between the text and the audio being generated prior to export as audiobook for the purpose of editing. Consider the following:

1.  Correcting pronunciations errors
2.  Manual manipulation and update the original text file that was used for generation
    - Systematic edits either per phrase/word with exact match or something like regular expression
4.  Bulk sentence regeneration and editting in case you want to regenerate audio for a sentence but not entire project
6.  Import/export projects so they can be edited at a later date if errors are found in generation
7.  Visual representation text boundaries (often sentences) to audio

This could be tied to a single interface if we could track audio generation to the underline text within the e-book. Imagine an editor like experience like WordPad.  Except as audiobook highlights the words and sentences currently in sync with the audio from generation. That way we can scrub through the e-book with audio similar to video editing and regenerate as needed.

**Comment by ROBERT-MCDOWELL** [comment]
it's an obvious feature indeed. would you work on it in the near future when version 2.0.0 will be released?

**Comment by LexiconCode** [comment]
Possibly parts of it, but there needs to be infrastructure in place to even attempt this. 2.0 looks like it starts to have some of the beginning features needed.

**Comment by JamesSwift** [comment]
From what I see of the files generates, it seems like one section of the VTT correlates to one sentence. Would it be possible to generate the new VTT, diff it to the old one, then only regenerate only the sections that changed (or were added/deleted)?

I could see the sentence cut points might drift after small changes, but that would only continue to the end of a chapter (I think). It could be a less-effort way of implementing this than a full overhaul perhaps? It wouldn't need a full new GUI interface.

**Comment by ROBERT-MCDOWELL** [comment]
what do you mean by VT?

**Reply by JamesSwift** [reply]
Sorry, I meant the .vtt file that is produced.

**Comment by ROBERT-MCDOWELL** [comment]
no the vtt is only the result of the conversion, it has nothing to do with pre-conversion. and btw the tmp folder is strictly used by E2A and must not be touched at all or a mess can happen. the only way to modify the conversion is only the original epub file.
until a preview will be developed.

**Comment by JamesSwift** [comment]
Thanks. Yeah, I don't modify the temp folder. I just saw the vtt and thought "if this matches the sentences, then it could be an easy way to diff them."
